test/lib/Benchmark.js

```diff
@@ -0,0 +1,182 @@
+/* globals XPCOMUtils, Services, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+// For the performance.now()
+const performance = Services.appShell.hiddenDOMWindow.performance;
+
+/*
+ * A simple benchmark framework that runs on top of jpm test
```

Comment:
I'd suggest writing here a list of what this framework is and isn't suitable for.  In particular, at least as it's currently written, it can't be used by itself to test "first-time" or "startup" costs (eg performance of the first time new tab is shown in a browser instance).